filebeat/input/journald/pkg/journalfield/matcher.go

```diff
@@ -118,3 +136,86 @@ func ApplyMatchersOr(j journal, matchers []Matcher) error {
 
 	return nil
 }
+
+// ApplySyslogIdentifierMatcher adds unit based filtering to the journal reader.
+func ApplyUnitMatchers(j journal, units []string, kernel bool) error {
+	for _, unit := range units {
+		matchers := [][]Matcher{
+			[]Matcher{
+				MustBuildMatcher("systemd.unit=" + unit),
+			},
+			[]Matcher{
+				MustBuildMatcher("message_id=fc2e22bc6ee647b6b90729ab34a250b1"),
+				MustBuildMatcher("journald.uid=1"),
+				MustBuildMatcher("journald.coredump.unit=" + unit),
+			},
+			[]Matcher{
+				MustBuildMatcher("journald.pid=1"),
+				MustBuildMatcher("journald.unit=" + unit),
+			},
+			[]Matcher{
+				MustBuildMatcher("journald.uid=1"),
+				MustBuildMatcher("journald.object.systemd.unit=" + unit),
+			},
```

Comment:
[Question]
Why those hardcoded message_id, pid and uid?

Reply:
The expressions are coming from the community Journalbeat: https://github.com/mheese/journalbeat/blob/e1e9376e9c01635c07135dd731956477c6d8cb2a/beater/unit.go#L103-L145
I will add more details to the function.